fe/fe-core/src/main/java/com/starrocks/sql/ast/ExpressionPartitionDesc.java

```diff
@@ -8,23 +8,41 @@
 
 public class ExpressionPartitionDesc extends PartitionDesc {
 
+    //must be column
     private SlotRef slotRef;
-    //expr in outputExpression
+    //must be type of SlotRef or FunctionCallRef
     private Expr expr;
+    // if expression is type of FunctionCallExpr, isFunc = true
+    // else isFunc = false
+    private boolean isFunction;
 
-    public ExpressionPartitionDesc(SlotRef slotRef) {
+    public ExpressionPartitionDesc(SlotRef slotRef, Expr expr, boolean isFunc) {
         this.slotRef = slotRef;
+        this.expr = expr;
```

Comment:
what is the value of type in PartitionDesc?

Reply:
it seems that you should init it.

Reply:
e.g. `create mv 1 partition by ss as select date_trunc('month',a) ss from table1`, expr will init in MaterializedViewAnalyzer